src/compatTest/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPluginTests.kt

```diff
@@ -889,6 +889,52 @@ class NexusPublishPluginTests {
         )
     }
 
+    @Test
+    fun `should not find staging repository by wrong description`() {
+        // given
+        writeDefaultSingleProjectConfiguration()
+        buildGradle.append("version='2.3.4-so staging repository is not found'")
```

Comment:
Could we do that when the value will actually be reused?

Reply:
If you mean to reuse the version only withing the method, I incline the current code is easier to maintain. I copy-pasted the actual output to the assertion message, and that is enough to update the assertion.

Note: the assertion message will likely need updates if the message is reworded. I incline it might happen, and every time it would require re-inserting the actual "non-matching" version

Reply:
I meant `2.3.4-so staging repository is not found` which is already reused could a `val` and "... ${x} ..." in the assertion message. I could be more precise. WDYT?

Btw, I have mixed feeling about that kind of big exact matching. Maybe checking just for 4-5 different element would be enough? Nevertheless, I'm not convinced, if my proposal would better in that case, so keep it as is and let's see how easy/hard it would be to maintain/read in practice.

Reply:
A bigger string gives much more information to the maintainer on the exact thing that was expected.
If you add an assert like  <<`2.3.4-so staging repository is not found` should be present in the output>>, then it would be unclear where and why the substring should appear.